src/Infrastructure/Common/Middlewares/SecurityMiddleware.cs

```diff
@@ -0,0 +1,97 @@
+﻿using DN.WebApi.Application.Identity.Exceptions;
+using DN.WebApi.Application.Identity.Interfaces;
+using DN.WebApi.Application.Settings;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using System.Collections.Concurrent;
+using System.Net;
+
+namespace DN.WebApi.Infrastructure.Middlewares;
+
+public class SecurityMiddleware : IMiddleware
+{
+    private static readonly ConcurrentDictionary<string, RequestFailed> _blackListIp = new();
```

Comment:
I think this belongs in a separate service (registered as a singleton) rather than as a static field.
There's 2 concerns here, 1) bookkeeping of the ip blacklist (IpBlacklistService maybe?) and 2) actually blocking users from abusing the system (the middleware).
I would also call the middleware something different... IpBlacklistMiddleware (just a suggestion). It also might warrant its own settings in stead of cramming them into MiddlewareSettings (these might probably be better split in LocalizationSettings and RequestLoggingSettings anyways).